src/store-api/src/region_engine.rs

```diff
@@ -27,6 +27,15 @@ use crate::metadata::RegionMetadataRef;
 use crate::region_request::RegionRequest;
 use crate::storage::{RegionId, ScanRequest};
 
+/// The result of setting readonly for the region.
+#[derive(Debug)]
+pub struct SetReadonlyResult {
+    /// Returns `last_entry_id` of the region if available(e.g., It's not available in file engine).
+    pub last_entry_id: Option<u64>,
```

Comment:
Use `store_api::logstore::entry::Id`?
```suggestion
    pub last_entry_id: Option<entry::Id>,
```